naga/src/front/spv/image.rs

```diff
@@ -656,10 +682,27 @@ impl<I: Iterator<Item = u32>> super::Frontend<I> {
             level,
             depth_ref,
         };
+        let image_sample_handle = ctx.expressions.append(expr, self.span_from_with_op(start));
+        let handle = if is_depth && depth_ref.is_none() {
+            let result_ty = self.lookup_type.lookup(result_type_id)?;
+            match ctx.type_arena[result_ty.handle].inner {
+                crate::TypeInner::Vector { size, .. } => {
```

Comment:
Do we have to check this here too? I thought all sampling functions return `vec4`s.

Reply:
I wasn't sure, so I decided to do the safe thing.

Reply:
I guess it doesn't hurt but might be misleading. I checked yesterday that all of SPIR-V's sampling functions that we handle all return `vec4`s, could you revert this?

Reply:
Of course 👍